sklearn/tests/test_isotonic.py

```diff
@@ -326,6 +329,30 @@ def test_isotonic_sample_weight():
     assert_array_equal(expected_y, received_y)
 
 
+def test_isotonic_sample_weight_centered():
+    x = [1, 2, 3, 4, 5, 6, 7]
+    y = [1, 41, 51, 1, 2, 5, 24]
+    sample_weight = [1, 2, 3, 4, 5, 6, 7]
+
+    expected_y_cir = [1, 4.7, 8.4, 12.1, 15.96, 19.98, 24]
```

Comment:
Where do these values come from? I smell a snake biting its tail:wink:
It would be nice if it were possible to construct values from which we know or can easily deduce the correct values, or take an example from the literature or a reference implementation.
In any case, comment from where these values come.

Reply:
These values were taken from the already-existing test for the non-centered scenario (see lines 321-329). Unfortunately I don't know where those came from originally, do you still recommend adding a comment where they come from? The expected outcomes were indeed calculated using this implementation, which is in a way a snake biting its tail.

I agree that it would be nice to have examples from a reference implementation, but unfortunately I couldn't find that. The original paper does not provide something along those lines. For the weighted samples (lines 321-353), I don't see an easily deduced or trivial case. For the unweighted test, there are easily deduced examples, which are indeed implemented in the function `test_isotonic_regression`.

Reply:
If there is an R implementation, we could write the R code as comments that produces those numbers - if it's only a few lines.

Reply:
That is a good idea, I have added the relevant R code snippets and outputs.